ee/ucp/kubernetes/layer-7-routing.md

```diff
@@ -19,87 +21,180 @@ Use an ingress controller when you want to:
-- a simple HTTP server, named `default-http-backend`,
-- an ingress controller, named `nginx-ingress-controller`, and
-- a service that exposes the app, named `ingress-nginx`.
+A dedicated namespace, for example, `nginx-controller`, is needed for all infrastructure deployment activities. You also need a service account to enable the ingress controller to work with the Kubernetes API. After creating a namespace and a service account, you must create an RBAC policy to only allow infrastructure deployments on the dedicated nodes that were labelled in the previous example.
 
-Navigate to the **Create Kubernetes Object** page, and in the **Object YAML**
-editor, paste the following YAML.
+To create a namespace and a service account, simply use the following [YAML file](config/ns-and-sa.yaml) and apply it via the CLI or the UI:
 
-```yaml
-apiVersion: apps/v1beta2
+```
+🐳  → cat ns-and-sa.yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: infra
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: nginx-ingress-service-account
+  namespace: infra
+```
+
+Then apply the namespace and service account via `kubectl`:
+
+```
+🐳  → kubectl apply -f ns-and-sa.yaml
+namespace "infra" created
+serviceaccount "nginx-ingress-service-account" created
+
+```
+
+### Step 3: Create an RBAC Policy
+
+Apply an RBAC role-binding for NGINX controller access to the API Server, as shown in the following [RBAC yaml config](config/ingress-rbac.yaml) example:
+
+
+```
+🐳  → cat ingress-rbac.yaml
+## Source: https://github.com/nginxinc/kubernetes-ingress/blob/master/deployments/rbac/rbac.yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: nginx-ingress-cluster-role
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - services
+  - endpoints
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  verbs:
+  - get
+  - list
+  - watch
+  - update
+  - create
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - list
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - create
+  - patch
+- apiGroups:
+  - extensions
+  resources:
+  - ingresses
+  verbs:
+  - list
+  - watch
+  - get
+- apiGroups:
+  - "extensions"
+  resources:
+  - ingresses/status
+  verbs:
+  - update
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: nginx-ingress-cluster-rb
+subjects:
+- kind: ServiceAccount
+  name: nginx-ingress-service-account
+  namespace: ingress
+roleRef:
+  kind: ClusterRole
+  name: nginx-ingress-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Then apply it with `kubectl`:
+
+```
+🐳  → kubectl apply -f ingress-rbac.yaml
+```
+
+### Step 4: Deploy NGINX Controller
+
+Note that following example uses hostPorts for controller ports. This exposes the host port ( selected in a high range using `hostPort: 38080`) directly into the nodes. 
```

Comment:
please add another note here : We are using port 38080 for HTTP and port 38443 for HTTPS. Make sure that your loadbalancer forwards to these ports on the nodes. You can change them as needed.

Reply:
Have added this with a couple minor wording changes. Thanks!